src/Sylius/Component/Core/StateResolver/OrderPaymentStateResolver.php

```diff
@@ -100,6 +100,21 @@ private function getTargetTransition(OrderInterface $order): ?string
             return OrderPaymentTransitions::TRANSITION_PARTIALLY_PAY;
         }
 
+        $authorizedPaymentTotal = 0;
+        $authorizedPayments = $this->getPaymentsWithState($order, PaymentInterface::STATE_AUTHORIZED);
+
+        foreach ($authorizedPayments as $payment) {
+            $authorizedPaymentTotal += $payment->getAmount();
+        }
+
+        if (0 < $authorizedPayments->count() && $authorizedPaymentTotal >= $order->getTotal()) {
+            return OrderPaymentTransitions::TRANSITION_AUTHORIZE_PAYMENT;
+        }
+
+        if ($authorizedPaymentTotal < $order->getTotal() && 0 < $authorizedPaymentTotal) {
+            return OrderPaymentTransitions::TRANSITION_PARTIALLY_AUTHORIZE_PAYMENT;
```

Comment:
Could you add test cases to `OrderPaymentStateResolverSpec` covering the code added here?

Reply:
It starts to be spooky here 👻 :( We need to refactor these state resolvers